samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst

```diff
@@ -0,0 +1,351 @@
+
+Reboot after download completion is handled by :file:`src/fota_support.c`, triggered by a call from :file:`src/connection.c`.
+
+`Device Shadow <nRF Cloud Device Shadows_>`_ updates are performed in :file:`src/connection.c`.
+
+In a real-world setting, it would suffice to have these two behaviors directly implemented in :file:`src/connection.c`.
+They are separated out in this sample for clarity.
+
+.. _nrf_cloud_mqtt_multi_services_sample_temperature:
+.. _nrf_cloud_mqtt_multi_services_sample_temperature_sensing:
+
+Temperature Sensing
+===================
+
+Temperature sensing is almost entirely implemented in :file:`src/temperature.c`.
+This includes generation of false temperature readings on the `Nordic nRF9160 DK`_, which does not have a built-in temperature sensor.
+
+However, use of the built-in temperature sensor of the `Nordic Thingy:91`_ requires a `devicetree overlay <Zephyr Devicetree Overlays_>`_, namely :file:`boards/thingy91_nrf9160_ns.overlay`, as well as a couple of enabled kconfig options (:kconfig:option:`CONFIG_SENSOR` and :kconfig:option:`CONFIG_BME680`).
+The devicetree overlay is automatically applied during compilation whenever the ``thingy91_nrf9160_ns`` boardname is selected.
+The required kconfig options are implicitly enabled by :ref:`CONFIG_TEMP_DATA_USE_SENSOR <CONFIG_TEMP_DATA_USE_SENSOR>`.
+
+.. _nrf_cloud_mqtt_multi_services_sample_location_tracking:
+.. _nrf_cloud_mqtt_multi_services_sample_location:
+
+Location Tracking
+=================
+
+All matters concerning location tracking are handled in :file:`src/location_tracking.c`.
+For the most part, this amounts to setting up a periodic location request, and then passing the results to a callback configured by :file:`src/application.c`.
+However, there are a couple of other tasks which must additionally be implemented:
+
+Firstly, the :ref:`location library <lib_location>` requires assisted GPS (AGPS) data in order have the fastest possible `time-to-first-fix <TTFF_>`_.
+The :ref:`location library <lib_location>` requests this data automatically, but has no way to automatically receive the response.
+Therefore, all MQTT messages received from `nRF Cloud`_ must be be passed directly through to the :ref:`nRF Cloud AGPS library <lib_nrf_cloud_agps>` to be checked for AGPS data.
+The :func:`location_agps_data_handler` function handles this task, and is sent MQTT messages directly from :file:`src/connection.c`
+
+Secondly, the GNSS module cannot obtain a lock without the antenna first being properly configured.
+This configuration is performed by :func:`gnss_antenna_configure`.
+
+.. _nrf_cloud_mqtt_multi_services_sample_test_counter:
+
+Test Counter
+============
+
+A test counter may optionally be enabled by enabling :ref:`CONFIG_TEST_COUNTER <CONFIG_TEST_COUNTER>`.
```

Comment:
good idea

Reply:
Thanks! It was supremely helpful during testing, and simple enough, I decided to keep it in the sample.